ocaml/xapi/repository_helpers.ml

```diff
@@ -1284,26 +1288,73 @@ let get_single_enabled_update_repository ~__context =
   in
   get_singleton enabled_update_repositories
 
-let with_access_token ~token ~token_id f =
-  match (token, token_id) with
-  | t, tid when t <> "" && tid <> "" ->
-      info "sync updates with token_id: %s" tid ;
-      let json = `Assoc [("token", `String t); ("token_id", `String tid)] in
-      let tmpfile, tmpch =
-        Filename.open_temp_file ~mode:[Open_text] "accesstoken" ".json"
+type client_auth =
+  | CdnTokenAuth (* remote *) of {token_id: string; token: string}
+  | NoAuth (* bundle *)
+  | PoolExtHostAuth (* remote_pool *) of {xapi_token: string}
```

Comment:
I think the plugin name should be part of the type. This can avoid the following in line 171:
```
(Uri.to_string uri, None, true, (NoAuth, ""), NoAuth)
```